recipes/sentry-crashpad/all/conanfile.py

```diff
@@ -44,22 +55,25 @@ def requirements(self):
 
     def validate(self):
         if self.settings.compiler.cppstd:
-            tools.check_min_cppstd(self, 11)
-
-        if tools.is_apple_os(self.settings.os):
-            # FIXME: add Apple support
-            raise ConanInvalidConfiguration("This recipe does not support Apple.")
-
-        if self.settings.os == "Linux":
-            if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < 5:
-                # FIXME: need to test for availability of SYS_memfd_create syscall (Linux kernel >= 3.17)
-                raise ConanInvalidConfiguration("sentry-crashpad needs SYS_memfd_create syscall support.")
+            # Set as required in crashpad CMake file.
+            # See https://github.com/getsentry/crashpad/blob/71bcaad4cf30294b8de1bfa02064ab629437163b/CMakeLists.txt#L67
+            tools.check_min_cppstd(self, 14)
 
-        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 15:
-            raise ConanInvalidConfiguration("Require at least Visual Studio 2017 (15)")
+        minimum_version = self._minimum_compilers_version.get(str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn("Compiler is unknown. Assuming it supports C++14.")
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration("Build requires support for C++14. Minimum version for {} is {}"
+                .format(str(self.settings.compiler), minimum_version))
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder)
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
```

Comment:
Patch needs to be applied in build

Reply:
It is applied here, because `no_copy_source` option is used. Because sources are not copied to the build folder, patches needs to be applied to the "original" sources. Moving them to build will require manual modification of patch properties to set correct paths.